tests/Features/ModelsResolverTest.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Features;
+
+use App\User;
+use Tests\TestCase;
+use Laravel\Scout\Builder;
+use Algolia\ScoutExtended\Searchable\ModelsResolver;
+
+final class ModelsResolverTest extends TestCase
```

Comment:
Feature testing means you put yourself on the mind of the developer using the FEATURES of the library. Developers will never use directly the models resolver. So you should only test methods exposed on the public api. In this case, the method resolving models is the `search`.

So you should add the following test to the `SeachTest::class`:

```php
    public function testSearchContainsMetadata(): void
    {
        $indexMock = $this->mockIndex(User::class);
        factory(User::class)->create();

        $indexMock->shouldReceive('search')->once()->andReturn([
            'hits' => [
                [
                    'objectID' => 'App\User::1',
                    '_highlightResult' => [],
                    '_rankingInfo' => [],
                ]
            ],
        ]);

        $scoutMetaData = Thread::search('')->get()->first()->scoutMetaData();

        $this->assertArrayHasKey('_highlightResult', $scoutMetaData);
        $this->assertArrayHasKey('_rankingInfo', $scoutMetaData);
    }

```

Reply:
I added the test and also performed some small changes to make it work. Let me know what you think :)